pyresample/kd_tree.py

```diff
@@ -352,20 +358,12 @@ def get_neighbour_info(source_geo_def, target_geo_def, radius_of_influence,
                                        reduce_data=reduce_data,
                                        nprocs=nprocs)
 
-            # Build result iteratively
-            if i == 0:
-                # First iteration
-                valid_output_index = next_voi
-                index_array = next_ia
-                distance_array = next_da
-            else:
-                valid_output_index = np.append(valid_output_index, next_voi)
-                if neighbours > 1:
-                    index_array = np.row_stack((index_array, next_ia))
-                    distance_array = np.row_stack((distance_array, next_da))
-                else:
-                    index_array = np.append(index_array, next_ia)
-                    distance_array = np.append(distance_array, next_da)
+            appendable_valid_output_index.append_row(next_voi)
+            appendable_index_array.append_row(next_ia)
+            appendable_distance_array.append_row(next_da)
+        valid_output_index = appendable_valid_output_index.to_array()
+        index_array = appendable_index_array.to_array()
+        distance_array = appendable_distance_array.to_array()
```

Comment:
Oh I like this code being cleaned up. Nice.

I'm not sure if you want to have this discussion in the PR or the issue, but maybe we should just pre-allocate the entire output size? We know that information before-hand don't we?

Reply:
Ah ok, I haven't seen this yet and so I answered in the issue, but I don't mind, wherever this dicussion is most suited.
But here quickly the tldr: some uses of `get_neighbour_info` don't fill up the entire output size, i.e. masked data - changes this might break user code